packages/rlc-common/src/helpers/nameUtils.ts

```diff
@@ -62,7 +62,7 @@ export const ReservedModelNames: ReservedName[] = [
   { name: "package", reservedFor: [NameType.Parameter] },
   { name: "private", reservedFor: [NameType.Parameter] },
   { name: "protected", reservedFor: [NameType.Parameter] },
-  { name: "public", reservedFor: [NameType.Parameter] },
+  { name: "public", reservedFor: [NameType.Parameter, NameType.Operation] },
```

Comment:
we will revisit all kinds of renaming when we handle https://github.com/Azure/autorest.typescript/issues/2214